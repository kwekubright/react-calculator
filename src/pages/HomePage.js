import React from 'react';

function HomePage() {
  return (
    <main>
      <section>
        <div className="container">
          <div className="welcome">
            <h2>Welcome to the Quoted App</h2>
            <p className="par">
              Sint eiusmod nisi commodo veniam irure mollit. Sunt ipsum nulla nisi eiusmod duis
              enim do proident et. Incididunt anim culpa dolore pariatur elit laboris esse
            </p>

            <p className="par">
              Ut incididunt aliqua dolor elit duis dolor Lorem nulla ea eiusmod eiusmod datat
              fugiat et excepteur esse.Labore occaecat veniam incididunt culpa magna dolore
              incididunt.
            </p>

            <p className="par">
              Aliquip aliquip eu reprehenderit adipisicing in Lorem duis amet. Eu elit culpa
              ea incididunt elit est nostrud ad ex adipisicing nulla dolor adipisicing.
              Pariatur laborum ipsum cillum Lorem deserunt sint dolor sit occaecat quis
              aliqua. Aliqua magna est aliquip duis qui quis pariatur. Excepteur quis sunt
              esse excepteur sunt aliqua exercitation commodo qui velit esse enim. Ea
              cupidatat nulla velit amet veniam irure ad excepteur.
            </p>
          </div>
        </div>
      </section>
    </main>
  );
}

export default HomePage;
