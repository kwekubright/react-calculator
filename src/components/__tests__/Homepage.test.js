import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import HomePage from '../../pages/HomePage';

describe('Homepage component test', () => {
  test('render the Homepage component', () => {
    const homepage = render(<HomePage />);
    expect(homepage).toMatchSnapshot();
  });

  test('check if the Homepage component rendered by checking if the h2 exist in the DOM', () => {
    render(<HomePage />);
    const h2Element = screen.getByText('Welcome to the Quoted App', { exact: false });
    expect(h2Element).toBeInTheDocument();
  });
});
