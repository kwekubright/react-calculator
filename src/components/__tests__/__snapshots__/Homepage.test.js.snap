// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage component test render the Homepage component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main>
        <section>
          <div
            class="container"
          >
            <div
              class="welcome"
            >
              <h2>
                Welcome to the Quoted App
              </h2>
              <p
                class="par"
              >
                Sint eiusmod nisi commodo veniam irure mollit. Sunt ipsum nulla nisi eiusmod duis enim do proident et. Incididunt anim culpa dolore pariatur elit laboris esse
              </p>
              <p
                class="par"
              >
                Ut incididunt aliqua dolor elit duis dolor Lorem nulla ea eiusmod eiusmod datat fugiat et excepteur esse.Labore occaecat veniam incididunt culpa magna dolore incididunt.
              </p>
              <p
                class="par"
              >
                Aliquip aliquip eu reprehenderit adipisicing in Lorem duis amet. Eu elit culpa ea incididunt elit est nostrud ad ex adipisicing nulla dolor adipisicing. Pariatur laborum ipsum cillum Lorem deserunt sint dolor sit occaecat quis aliqua. Aliqua magna est aliquip duis qui quis pariatur. Excepteur quis sunt esse excepteur sunt aliqua exercitation commodo qui velit esse enim. Ea cupidatat nulla velit amet veniam irure ad excepteur.
              </p>
            </div>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <main>
      <section>
        <div
          class="container"
        >
          <div
            class="welcome"
          >
            <h2>
              Welcome to the Quoted App
            </h2>
            <p
              class="par"
            >
              Sint eiusmod nisi commodo veniam irure mollit. Sunt ipsum nulla nisi eiusmod duis enim do proident et. Incididunt anim culpa dolore pariatur elit laboris esse
            </p>
            <p
              class="par"
            >
              Ut incididunt aliqua dolor elit duis dolor Lorem nulla ea eiusmod eiusmod datat fugiat et excepteur esse.Labore occaecat veniam incididunt culpa magna dolore incididunt.
            </p>
            <p
              class="par"
            >
              Aliquip aliquip eu reprehenderit adipisicing in Lorem duis amet. Eu elit culpa ea incididunt elit est nostrud ad ex adipisicing nulla dolor adipisicing. Pariatur laborum ipsum cillum Lorem deserunt sint dolor sit occaecat quis aliqua. Aliqua magna est aliquip duis qui quis pariatur. Excepteur quis sunt esse excepteur sunt aliqua exercitation commodo qui velit esse enim. Ea cupidatat nulla velit amet veniam irure ad excepteur.
            </p>
          </div>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
